### PR TITLE
more sonarlint

### PR DESCRIPTION
**Pull request by neheb**
Signed-off-by: Rosen Penev <rosenp@gmail.com>

### CODE REVIEW

**Comment by neheb** [comment]
that find_if lambda though...

Need to add a constexpr library to simplify with find().

File: src/content/scripting/playlist_parser_script.cc [hunk 1]
```diff
@@ -94,9 +94,9 @@ duk_ret_t jsReadln(duk_context* ctx)
     }
 
     try {
-        auto line = self->readLine();
-        duk_push_string(ctx, line.first.c_str());
-        return line.second ? 1 : 0;
+        auto [line, trimmed] = self->readLine();
+        duk_push_string(ctx, line.c_str());
+        return trimmed;
```
Comment (KarlStraussberger) [inline src/content/scripting/playlist_parser_script.cc:99]: We should not implicitely convert `bool` to `duk_ret_t`

Comment (neheb) [reply src/content/scripting/playlist_parser_script.cc]: How about now?

Comment (KarlStraussberger) [reply src/content/scripting/playlist_parser_script.cc]: I'd leave the original `? 1 : 0`. You see what you get.

Comment (neheb) [reply src/content/scripting/playlist_parser_script.cc]: OK.

**Pull request merged**